test/unicode/checkstring.jl

```diff
@@ -89,11 +89,7 @@ try
         @test_throws UnicodeError Base.checkstring(UInt8[byt,0x80,0x80,0xc0])
     end
 
-    # Long encoding of 0x01
-    @test_throws UnicodeError utf8(b"\xf0\x80\x80\x80")
-    # Test ends of long encoded surrogates
-    @test_throws UnicodeError utf8(b"\xf0\x8d\xa0\x80")
-    @test_throws UnicodeError utf8(b"\xf0\x8d\xbf\xbf")
```

Comment:
For the new String type, it will accept any data, valid UTF-8 or not – but doesn't yet, so we're in a slightly awkward intermediate state. I promise I will add lots of tests for all of that.